merqube_client_lib/templates/equity_baskets/util.py

```diff
@@ -138,6 +189,30 @@ def load_template(
     return client, template, index_info, inner_spec
 
 
+def _configure_report(template: IndexDefinitionPost, email_list: list[str] | None = None) -> None:
+    """
+    confifugre index reports
+    """
+    updated_reports = []
+    if not email_list or not template.run_configuration or not template.run_configuration.index_reports:
+        logger.warning("Warning, index has no reports to configure or has not been given an email list")
+        return
+
+    for ind, report in enumerate(template.run_configuration.index_reports):
+        updated_report = cast(IndexReport, deepcopy(report))
+        assert updated_report.program_args
+        diss_config = json.loads(updated_report.program_args["diss_config"].replace('\\"', '"'))
+
+        if not (dc := diss_config.get("email")) or len(dc) > 1:
+            logger.warning(f"Warning, report {ind} has no email config or has more than one config")
+            continue
+
+        diss_config["email"][0]["recipients"] = email_list
+        updated_report.program_args["diss_config"] = json.dumps(diss_config).replace('"', '\\"')
```

Comment:
To do that we need to update how indexapi generates dags. Right it allows you to pass in an arguments `dict[str, str]` which it sets as arguments passed to the dag task. If the value is a `dict` it will generate an invalid dag. If we change this so that it stringifies the `dict` before adding it as an argument to the dag then we wouldn't need to do that here.